src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -6,147 +6,131 @@
+        }
 
-            if (packageFeed == null)
+        private async ValueTask InitializeAsync(INuGetSearchService searchService = null)
+        {
+            _searchFilter = new SearchFilter(includePrerelease: _includePrerelease)
             {
-                throw new ArgumentNullException(nameof(packageFeed));
-            }
-            _packageFeed = packageFeed;
-            _recommenderPackageFeed = recommenderPackageFeed;
+                SupportedFrameworks = await _context.GetSupportedFrameworksAsync()
+            };
 
-            _searchText = searchText ?? string.Empty;
-            _includePrerelease = includePrerelease;
+            _serviceBroker = await BrokeredServicesUtilities.GetRemoteServiceBrokerAsync();
+            _serviceBroker.AvailabilityChanged += OnAvailabilityChanged;
+            _searchService = searchService ?? await GetSearchServiceAsync(CancellationToken.None);
         }
 
-        public async Task<int> GetTotalCountAsync(int maxCount, CancellationToken cancellationToken)
+        private void OnAvailabilityChanged(object sender, BrokeredServicesChangedEventArgs e)
         {
-            // Go off the UI thread to perform non-UI operations
-            await TaskScheduler.Default;
-
-            ActivityCorrelationId.StartNew();
-
-            int totalCount = 0;
-            ContinuationToken nextToken = null;
-            do
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
-                var searchResult = await SearchAsync(nextToken, cancellationToken);
-                while (searchResult.RefreshToken != null)
-                {
-                    searchResult = await _packageFeed.RefreshSearchAsync(searchResult.RefreshToken, cancellationToken);
-                }
-                totalCount += searchResult.Items?.Count() ?? 0;
-                nextToken = searchResult.NextToken;
-            } while (nextToken != null && totalCount <= maxCount);
+                _searchService?.Dispose();
+                _searchService = await GetSearchServiceAsync(CancellationToken.None);
+            }).PostOnFailure(nameof(PackageItemLoader), nameof(OnAvailabilityChanged));
+        }
+
+        private async ValueTask<INuGetSearchService> GetSearchServiceAsync(CancellationToken cancellationToken)
+        {
+#pragma warning disable ISB001 // Dispose of proxies
+            INuGetSearchService searchService = await _serviceBroker.GetProxyAsync<INuGetSearchService>(NuGetServices.SearchService, cancellationToken);
+#pragma warning restore ISB001 // Dispose of proxies
+            Assumes.NotNull(searchService);
+            return searchService;
+        }
 
-            return totalCount;
+        public async Task<int> GetTotalCountAsync(int maxCount, CancellationToken cancellationToken)
+        {
+            return await _searchService.GetTotalCountAsync(maxCount, _context.Projects, _packageSources, _searchFilter, _itemFilter, cancellationToken);
```

Comment:
The old `GetTotalCountAsync(...)` moved onto a thread pool thread first.  Why not here?